.github/workflows/publish.yml

```diff
@@ -0,0 +1,45 @@
+name: Publish
+on:
+  push:
+      branches: [master]
+
+jobs:
+  build:
+    uses: ./.github/workflows/build-and-test.yml
+  publish:
+    needs: build
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/semversioner
+    permissions:
+        id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+    
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+
+      - name: Install Poetry 1.8.2
+        uses: abatilo/actions-poetry@v3.0.0
+        with:
+          poetry-version: '1.8.2'
+
+      - name: Install dependencies
+        shell: bash
+        run: poetry install
+
+      - name: Build Distributable
+        shell: bash
+        run: poetry build
+
+      - name: Publish package distributions to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
```

Comment:
It is possible to use `poetry publish` here instead of relying on the github action?